backends/tfhe-cuda-backend/cuda/include/integer.h

```diff
@@ -593,20 +608,58 @@ template <typename Torus> struct int_radix_lut {
                              host_free_on_stream_callback, h_lwe_indexes);
   }
 
-  Torus *get_lut(size_t ind) {
+  // Return a pointer to idx-ith lut at gpu_index's global memory
+  Torus *get_lut(uint32_t gpu_index, size_t idx) {
+    auto lut = lut_vec[gpu_index];
+    size_t lut_size = (params.glwe_dimension + 1) * params.polynomial_size;
+
     assert(lut != nullptr);
-    return &lut[ind * (params.glwe_dimension + 1) * params.polynomial_size];
+    return &lut[idx * lut_size];
   }
 
-  Torus *get_lut_indexes(size_t ind) { return &lut_indexes[ind]; }
+  // Return a pointer to idx-ith lut indexes at gpu_index's global memory
+  Torus *get_lut_indexes(uint32_t gpu_index, size_t ind) {
+    auto lut_indexes = lut_indexes_vec[gpu_index];
+    return &lut_indexes[ind];
+  }
+
+  // Broadcast luts from gpu src_gpu_idx to all active gpus
+  void broadcast_lut(cudaStream_t *streams, uint32_t *gpu_indexes,
```

Comment:
Maybe it would be good to have two separate entry points to broadcast lut and lut_indexes separately? It's more work though and we might not gain much.

Reply:
We can open an issue to do it later maybe

Reply:
We can open an issue to do it later maybe

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/609